File: Part2-Regression/Section 11 - Random Forest Regression/R/Random_forest_regression.R
#Random forest resgression

#importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#Fitting Random forest Regression to the dataset
# install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor = randomForest(x = dataset[1],
                         y = dataset$Salary,
                         ntree = 300)

#predicting a new result
y_pred = predict(regressor,data.frame(Level =6.5))

#Visualising the plot 
library(ggplot2)
x_grid = seq(min(dataset$Level),max(dataset$Level),0.1)
ggplot() + 
  geom_point(aes(x = dataset$Level, y=dataset$Salary),
             color='red') +
  geom_line(aes(x = x_grid, y=predict(regressor,newdata = data.frame(Level = x_grid)))
            ,color = 'blue') + 
  ggtitle('Truth or bluff(Random Forest Regression)')+
  xlab('Level')+
  ylab('Salary')

             
  

File: Part2-Regression/Section 11 - Random Forest Regression/Python/random_forest_regression.py
# Random Forest Regression

# Importing the libraries

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values

#Fitting Random Forest Regressioin to the dataset
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 1000,criterion='mse',random_state=0)
regressor.fit(X,y)

#predicting the new result
y_pred = regressor.predict([[6.5]])

#Visualising the result
X_grid = np.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color = 'red')
plt.plot(X_grid,regressor.predict(X_grid),color='blue')
plt.title('Truth or bluff(Random Forest Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()